### PR TITLE
remote actions

### LINKED ISSUE #31: Add actions from remote API to teaser lists
## Acceptance Criteria

- [x] I can see actions teasers on the `/guides` page#
- [x] An actions teaser contains a title
- [x] An actions teaser contains a short description
- [x] An actions teaser links to its canonical home
- [ ] An actions teaser can be searched for (to be implemented as part of #119)

## Description
Make a plan about how we will use the actions data.

The existing JSON API for the actions content is available here: [https://www.wildlifetrusts.org/jsonapi/node/action](https://www.wildlifetrusts.org/jsonapi/node/action) 

- as discussed any reuse of this content will require canonical links to point to original pages on national site to avoid SEO penalties. 

### We need
- check external API provides data used in teaser list
- to include actions guides in our search
- provide links to the content

## Plan
- [ ] make a HTTP request and update the model
- [ ] display as part of teaser list
- [ ] link to external resource

### PR DESCRIPTION
**Pull request by aaaaargZombies**
Fixes #31 

## Description

- Makes http request to static json file to mock the API end point, we will need to update this when we are given CORS access see this issue #135
- Use `Page.Shared.View.viewGuideTeaserList` to generate list of actions based on our guide content and remote actions
- Apologies for the stream of WIP  and broken state commits, I'll squash this one in or do an extensive rebase depending on reviewer preference.

## Questions

- There are some hard coded values in here, should I set up [elm-constants](https://www.npmjs.com/package/elm-constants) to allow this to be more easily configured and documented.
- is `Page.Shared.View.viewGuideTeaserList` the correct component for this or do we want to write something specific for the `/guides` route
- There is a comment that suggests refactoring the  `GuideTeaser`  struct to use a `Url.Url` type now we have external links. I have not done this with the intention of taking the path of least resistance, but I could do.
- I've had to move `GuideTeaser` and `Image` into their own `Page.Shared` module to deal with an import cycle, is this the best place?
- Should I add a combined local / external `List GuideTeaser` field or is it better to always build it in the place it gets used.

@geeksforsocialchange/developers

### CODE REVIEW

File: src/Page/Story/Data.elm [hunk 1]
```diff
@@ -13,9 +14,9 @@ type alias Story =
     , slug : String
     , maybeLocation : Maybe String
     , maybeGroupOrIndividual : Maybe String
-    , maybeImages : Maybe (List Page.Shared.View.Image)
+    , maybeImages : Maybe (List Image)
     , fullTextMarkdown : String
-    , relatedGuideList : List Page.Shared.View.GuideTeaser
+    , relatedGuideList : List GuideTeaser
```
Comment (aaaaargZombies) [inline src/Page/Story/Data.elm:19]: request change: show import location for internal code rather than destructuring

File: src/Message.elm [hunk 2]
```diff
@@ -11,3 +13,4 @@ type Msg
     | CookieSettingsButtonClicked
     | CookiesAccepted
     | CookiesDeclined
+    | GotActions (Result Http.Error (List GuideTeaser))
```
Comment (r-ferrier) [inline src/Message.elm:16]: prefer Page.Shared.GuideTeaser

File: src/Page/Guide/Data.elm [hunk 3]
```diff
@@ -17,7 +18,7 @@ type alias Guide =
     , maybeVideo : Maybe Page.Shared.View.VideoMeta
     , maybeAudio : Maybe Page.Shared.View.AudioMeta
     , relatedStoryList : List Page.Shared.View.StoryTeaser
-    , relatedGuideList : List Page.Shared.View.GuideTeaser
+    , relatedGuideList : List GuideTeaser
```
Comment (r-ferrier) [inline src/Page/Guide/Data.elm:21]: prefer Page.Shared.GuideTeaser

File: src/Page/Guide/Data.elm [hunk 4]
```diff
@@ -89,7 +90,7 @@ slugToUrl slug =
 teaserListFromGuideDict :
     Language
     -> Dict String Guide
-    -> List Page.Shared.View.GuideTeaser
+    -> List GuideTeaser
```
Comment (r-ferrier) [inline src/Page/Guide/Data.elm:93]: prefer Page.Shared.GuideTeaser

File: src/Page/Shared.elm [hunk 5]
```diff
@@ -0,0 +1,15 @@
+module Page.Shared exposing (GuideTeaser, Image)
+
+
+type alias Image =
+    { src : String, alt : String }
+
+
+type alias GuideTeaser =
```
Comment (ivan-kocienski-gfsc) [inline src/Page/Shared.elm:8]: I think this should be in `src/Page/Guides/Data.elm` as it is tightly coupled to the full Guide resource.

File: src/Page/Shared.elm
Comment (r-ferrier) [inline src/Page/Shared.elm:1]: I understand where you're going with this but it feels a bit weird having this & also a shared folder - was there any reason why these couldn't live in shared/data?

File: src/Page/Shared.elm (hunk 5 above)
Comment (aaaaargZombies) [reply src/Page/Shared.elm:8]: sadly that will cause a import loop but does make more sense as a category

File: src/Page/Shared.elm [hunk 6]
```diff
@@ -0,0 +1,15 @@
+module Page.Shared exposing (GuideTeaser, Image)
+
+
+type alias Image =
```
[the remark this answers is not in the document]
Comment (aaaaargZombies) [reply src/Page/Shared.elm:4]: I'll try this and see what happens

[the remark this answers is not in the document]
Comment (aaaaargZombies) [reply src/Page/Shared.elm:4]: oops no this causes an import cycle

**Review by r-ferrier — APPROVED** [review]
I think this works and looks overall good, there's something about two things called `shared` at the same level in the tree though that feels like something is not quite right. If this is not possible to change because of cyclical references then I think it's probably better to add for now and make a new issue to address the potential refactoring we need to do, as it feels like a bigger question. The other changes I requested are just naming conventions.

I'm approving so you can make the call yourself on what to do with this!

I also think you should just make a new issue for the comment you have about refactoring a GuideTeaser struct to use a Url.Url type now we have external links

I think keeping local/external guides separate is fine, I can only see the list in the design in two places & it looks different in both so we may need to confirm anyway if both are to contain the same content.

File: src/Page/Shared.elm
Comment (r-ferrier) [reply src/Page/Shared.elm:1]: have read above - understood!

**Pull request merged**